lib/block-supports/typography.php

```diff
@@ -276,55 +278,74 @@ function gutenberg_get_typography_value_and_unit( $raw_value, $options = array()
 		return null;
 	}
 
+	$defaults = array(
+		'coerce_to'         => '',
+		'root_size_value'   => 16,
+		'skip_unit_parsing' => false,
+		'acceptable_units'  => array( 'rem', 'px', 'em' ),
+	);
+
+	$options           = wp_parse_args( $options, $defaults );
+	$skip_unit_parsing = true === $options['skip_unit_parsing'];
+
 	// Converts numeric values to pixel values by default.
-	if ( is_numeric( $raw_value ) ) {
+	if ( is_numeric( $raw_value ) && ! $skip_unit_parsing ) {
 		$raw_value = $raw_value . 'px';
 	}
 
-	$defaults = array(
-		'coerce_to'        => '',
-		'root_size_value'  => 16,
-		'acceptable_units' => array( 'rem', 'px', 'em' ),
-	);
-
-	$options = wp_parse_args( $options, $defaults );
+	// For floats coerced to strings, locales can use either a comma (,) or dot (.) as decimal separators.
+	// If the locale isn't using a dot, we need to replace the locale separator with a dot for CSS rules to be valid.
+	$local_info        = localeconv();
```

Comment:
I did try the following in all the typography functions: 

```php
	$current_locale = setlocale(LC_NUMERIC, 0 );
	setlocale(LC_NUMERIC, 'C');

        // do some work

	setlocale(LC_NUMERIC, $current_locale );
```

It works fine, but I'm not 100% sure if we should be flipping locales like this, especially `n` times?

I had some success with `json_encode` as well. If that ends up being simpler, I can switch.

Needs more testing.

Reply:
Also, from what I've read, there is an argument that the server should be handling the behaviour here. If that argument holds, then this PR isn't required at all.